coreos-ostree-importer/coreos_ostree_importer.py

```diff
@@ -48,14 +49,34 @@
     "compose": "/mnt/koji/compose/ostree/repo",
 }
 
-
 class Consumer(object):
     def __init__(self):
-        # Check the possible repos to make sure they exist
-        for path in KNOWN_OSTREE_REPOS.values():
+        # Do sanity checks on the repos
+        for repo, path in KNOWN_OSTREE_REPOS.items():
+
+            # Check the repo to make sure it exists
             if not ostree_repo_exists(path):
                 raise Exception(f"OSTree repo does not exist at {path}")
 
+            # Sanity check the repo to make sure all directories in the repo
+            # are group writable. https://pagure.io/releng/issue/8811#comment-616490
+            # Wrap in try/except so we can hopefull see the logs before openshift
+            # deletes the pod
+            try:
+                assert_dirs_group_writable(repo)
+            except Exception as e:
+                logger.error(f"Exception detected: {str(e)}")
+                logger.info(f"Sleeping for 30s")
+                time.sleep(30)
+                raise(e)
+
+        # Configure a umask of 0002 which will allow for the group permissions
```

Comment:
Hmm, feels odd to do this from a class which gets run as part of another app. This is normally done early on in an app startup to avoid issues with multi-threading. A safer approach is to make this part of a tiny init script that does `umask` and then `exec`s `fedora-messaging`. Or could just do it in the Dockerfile too, e.g.:

```
CMD ["sh", "-c", "umask 0002 && exec fedora-messaging consume --callback=coreos_ostree_importer:Consumer"]
```

Reply:
Part of this PR configures the container itself (via /etc/profile.d) to have a umask of 0002. Doing it here too was was mostly a precaution in case the app ever got run outside of the container for whatever reason. We can probably just delete this.

Reply:
Hmm, I think that's just an artifact of how Docker runs containers that don't use brackets (so it shells out to Bash, which sources the profile). Not sure if it's something we'd actually want to rely on.

Reply:
I put it in `/etc/profile.d/umask-for-openshift.sh` with a reason (partly so anyone running `oc rsh` will also get `umask 0002`) so we definitely need it there. Right now I also have it in `coreos_ostree_importer.py` (as good measure because the ` /etc/profile.d/umask-for-openshift.sh` is effective enough to apply here too) but you think it shouldn't be there and you'd rather have it as `CMD ["sh", "-c", "umask 0002 && exec fedora-messaging consume --callback=coreos_ostree_importer:Consumer"]` in the Dockerfile itself because we shouldn't rely on the *non `[]` arguments to command being run as a shell* behavior?

Reply:
I think having it in `/etc/profile.d` makes sense as you said to have it baked in the image. But for the app itself, given that the proper `umask` is critical here, it seems safer to have the app do that itself. The problem here though is that we're not actually in control of the app being launched.

Anyway, let's keep the `os.umask` here for now, and then in a follow-up we change the entrypoint to the app be a simple wrapper shell/python script that just sets umask and `exec`s `fedora-messaging`?

Reply:
added a `coreos-ostree-importer-wrapper` script that should take care of the concerns here